### ISSUE TITLE
pre trained weights

### ISSUE THREAD

**Issue by Padmini588** [SanoScience/FetalNet#4]
when i tried to load the pretrained weights with number of classes as 4 i get an error of shape mismatch. However when i use number of classes as 1 it doesnt throw an error. I have confusion between the details specified in the paper and pre trained weights . Are pre- trained weights based on single class segementation.

**Comment by simongeek** [comment]
Hi,

The number of classes = 4 is for classification task, and number of output channels = 1 for segmentation 

`out = self.out_layer(torch.cat((s_1, s_2, s_3, s_4), 1))  # [1, 1, 224, 224])`.

**Comment by Padmini588** [comment]
So can it segment only one class, why the does paper say the model is trained to segment 4 classes

**Comment by Padmini588** [comment]
where can i find the code used for training

**Comment by simongeek** [comment]
We use one class to segment all structures as binary (loss for segmentation), but we have four classes for classification (loss for classification). In the end, we can differentiate each structure: background, fetal abdomen, fetal head, and fetal femur.

We did not release the training code.

Best

**Comment by Padmini588** [comment]
Thank u for clarifying it.
